### PR TITLE
Add replica option to coderd

### PR DESCRIPTION
**Pull request by f0ssel**
todo:
- [ ] decide if we wanna call the `replica` option something else so it's no confused with `coderd.replicas` field

### CODE REVIEW

File: templates/coderd.yaml [hunk 1]
```diff
@@ -156,11 +162,17 @@ spec:
...
+              {{- if not .Values.coderd.replica.enable }}
               echo Waiting on db;
               PGSSLMODE="${DB_SSL_MODE:-prefer}" /wait_postgres.sh --host="$DB_HOST" --port="$DB_PORT" -U "$DB_USER";
               echo Starting entrypoint.sh;
               # Pass signals down to the envmanager.
               exec /entrypoint.sh coderd run
+              {{- else }}
+              echo Starting replica;
+              # Pass signals down to the envmanager.
+              exec /entrypoint.sh coderd replica --main-url {{ .Values.coderd.replica.primaryURL }}
```
Comment (deansheather) [inline templates/coderd.yaml:174]:
```suggestion
              exec /entrypoint.sh coderd replica --main-url {{ .Values.coderd.replica.primaryURL | quote }}
```

File: values.yaml [hunk 2]
```diff
@@ -391,3 +391,8 @@ serviceAccount:
   labels: {}
 
 deploymentAnnotations: {}
+
+coderd:
```
Comment (deansheather) [inline values.yaml:395]: Comments

**Review by deansheather — APPROVED** [review]
When will you add ecdsa keys?

**Comment by f0ssel** [comment]
I'll add the key in this PR as well

**Pull request merged**